var/spack/repos/builtin/packages/py-pycompadre/package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPycompadre(PythonPackage):
+    """The Compadre Toolkit provides a performance portable solution for the
+    parallel evaluation of computationally dense kernels. The toolkit
+    specifically targets the Generalized Moving Least Squares (GMLS) approach,
+    which requires the inversion of small dense matrices. The result is a set
+    of weights that provide the information needed for remap or entries that
+    constitute the rows of some globally sparse matrix.
+    """
+
+    homepage    = 'https://github.com/SNLComputation/compadre'
+    git         = 'https://github.com/SNLComputation/compadre.git'
+    url         = 'https://github.com/SNLComputation/compadre/archive/v1.3.0.tar.gz'
+    maintainers = ['kuberry']
+
+    version('master',  branch='master', preferred=True)
+
+    variant('trilinos', default=False, description='Use Kokkos from Trilinos')
+    variant('debug', default=False, description='Enable runtime safety and debug checks')
+    variant('extremedebug', default=False, description='Enable entry by entry checks')
+
+    depends_on('cmake@3.10.0:', type='build')
+    depends_on('python@3.4:', type=('build', 'link', 'run'))
+    depends_on('py-pip', type=('build', 'link', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-cython@0.23:', type='build')
+    depends_on('trilinos@13.2:', when='+trilinos')
+
+    with when('~debug'):
+        conflicts('+extremedebug')
```

Comment:
Can we have a single valued variant for debug instead of 2 variants and a conflict?

Reply:
Hi @alalazo Thanks for the suggestion. `extremedebug` is quite different from `debug` (it checks every entry of every vector/matrix for Nans in addition to a lot of other things). I will remove the conflict and handle it in the CMake. I can switch to a `debug` variant with options 0,1,2 corresponding to different debug levels. This will work for installation. 

I'm noticing that if I use this, though, and install three different versions with `spack install py-pycompadre debug=0`, `spack install py-pycompadre debug=1`, and `spack install py-pycompadre debug=2`, then saying `spack load py-pycompadre` isn't sufficient to distinguish which installation I'm intending (default for debug is 0). 

Are defaults for variants not used when calling `spack load [package]`?